backend/internal/service/app_config_service.go

```diff
@@ -202,30 +213,52 @@ func (s *AppConfigService) UpdateAppConfig(input dto.AppConfigUpdateDto) ([]mode
 		return nil, &common.UiConfigDisabledError{}
 	}
 
-	tx := s.db.Begin()
+	// Extract values outside of the lock
 	rt := reflect.ValueOf(input).Type()
 	rv := reflect.ValueOf(input)
 
-	var savedConfigVariables []model.AppConfigVariable
+	// Prepare the updates outside the lock
+	var updates []struct {
+		Key   string
+		Value string
+	}
+
 	for i := 0; i < rt.NumField(); i++ {
 		field := rt.Field(i)
 		key := field.Tag.Get("json")
 		value := rv.FieldByName(field.Name).String()
 
-		// If the emailEnabled is set to false, disable the emailOneTimeAccessEnabled
-		if key == s.DbConfig.EmailOneTimeAccessEnabled.Key {
+		// Apply special case logic for email settings
+		emailOneTimeAccessKey := ""
+		s.mu.RLock() // Short read lock just to get the key
+		emailOneTimeAccessKey = s.DbConfig.EmailOneTimeAccessEnabled.Key
+		s.mu.RUnlock()
```

Comment:
i strongly suggest instead of doing this, create setter/getter which locks/unlocks within the function.

If you patch this race condition like this, it might create in the future more bugs caused by deadlocks.